frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -30,29 +31,32 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   dirty,
   namespace,
   projects,
-}) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
-    <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection />}
-    {values.type === EventSources.KafkaSource && <KafkaSourceSection />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    {values.type === EventSources.KafkaSource && <AdvancedSection />}
-    <FormFooter
-      handleReset={handleReset}
-      errorMessage={status && status.submitError}
-      isSubmitting={isSubmitting}
-      submitLabel="Create"
-      sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
-      resetLabel="Cancel"
-    />
-  </Form>
-);
+}) => {
+  return (
+    <Form className="co-deploy-image" onSubmit={handleSubmit}>
+      <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
+      <AsyncComponent
+        loader={eventSourceSectionLoader(values.type)}
+        eventSourceType={values.type}
+      />
+      {isKnownEventSourceSelected(values.type) && <SinkSection namespace={namespace} />}
```

Comment:
Why do you have two same checks here? Merge them together.

Reply:
done